apphome/stats.go

```diff
@@ -70,3 +71,50 @@ func messagesAnalyzed(userIdFilt expression.ConditionBuilder) int {
 	}
 	return val
 }
+
+func userProj()	expression.ProjectionBuilder {
+	return expression.NamesList(expression.Name("user_id"))
+}
+
+func UsersWithBadQualityMessages() ([]string, error) {
+	filter := expression.And(sinceBeginningOfQuarterFilt(), badQualityFilt())
+	proj := userProj()
+	expr, buildErr := expression.NewBuilder().WithProjection(proj).WithFilter(filter).Build()
+	if buildErr != nil {
+		log.Println("Got error building expression:")
+		log.Println(buildErr.Error())
+		return []string{}, buildErr
+	}
+
+	input := dynamodb.ScanInput{
+		ExpressionAttributeNames:  expr.Names(),
+		ExpressionAttributeValues: expr.Values(),
+		FilterExpression:          expr.Filter(),
+		ProjectionExpression:	   expr.Projection(),
+		TableName:                 aws.String(os.Getenv("DYNAMODB_TABLE")),
+	}
+	type record struct {
+		UserId string `dynamodbav:"user_id"`
+	}
+	var records []record
```

Comment:
I know it's doing more than necessary, but what about reusing the existing db.Message struct that should already model the whole DDB item ?
I'm aware that 
- it's less efficient, but it would allow reusing the same existing code than having to write a "partial" struct for unmarshalling
- even if the DDB scan will only return user IDs because of your projection, it's still okay to build the Message struct with the UserId and rely on default values ?

I'm not really sure about what's the usual standards in Go so it's more like a question than a change request

Reply:
Removed